pandas_gbq/gbq.py

```diff
@@ -1007,6 +1007,12 @@ def to_gbq(
         raise NotFoundException(
             "Invalid Table Name. Should be of the form 'datasetId.tableId' "
         )
+    has_credentials = (
+        credentials is not None
+        and credentials.__class__.__name__.endswith("Credentials")
```

Comment:
Is there an approach here that is more robust to the name of the class? Do all Credentials classes have a common base we could test for?

Reply:
Yeah, they all have https://google-auth.readthedocs.io/en/latest/reference/google.auth.credentials.html#google.auth.credentials.Credentials as a base class.